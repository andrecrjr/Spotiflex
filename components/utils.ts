export const spotifyAuth = async (): Promise<void> => {
  const body = 'grant_type=client_credentials';
  const response = await fetch('https://accounts.spotify.com/api/token', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
      Authorization: `Basic ${Buffer.from(
        `09cc6d07722546fdbb6f06e4e9161f90:${process.env.SPOTIFY_SECRET_KEY}`
      ).toString('base64')}`,
    },
    body: body,
  });
  const data = await response.json();
  return data;
};

export function msToTime(duration: number) {
  var milliseconds = Math.floor((duration % 1000) / 100),
    seconds = Math.floor((duration / 1000) % 60),
    minutes = Math.floor((duration / (1000 * 60)) % 60),
    hours = Math.floor((duration / (1000 * 60 * 60)) % 24);

  let newMinutes = minutes < 10 ? '0' + minutes : minutes;
  let newSeconds = seconds < 10 ? '0' + seconds : seconds;

  return newMinutes + ':' + newSeconds;
}
