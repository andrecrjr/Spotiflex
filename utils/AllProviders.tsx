
import Layout from "@/components/Layout"
import { UserPlaylistWrapper } from "@/components/contextWrapper/UserPlaylist"
import TransactionPage from "@/components/TransactionPage"
import { render, RenderOptions } from "@testing-library/react"
import { ReactElement } from "react"

const AllProviders = ({children}) =>{
    return (
        <UserPlaylistWrapper>
            <Layout>
                <TransactionPage>{children}</TransactionPage>
            </Layout>
        </UserPlaylistWrapper>
    )
  }

export const renderWithWrappers = (component:ReactElement, options:RenderOptions={}) =>{
    return (
        render(component, {wrapper:AllProviders, ...options})
    )
}