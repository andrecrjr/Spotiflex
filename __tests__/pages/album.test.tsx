import { renderWithWrappers } from '../../utils/AllProviders';
import Album from '@/pages/album/[id]';
import { screen } from '@testing-library/react';
import albumMock from '@/mocks/oneHotMinuteAlbumMock.json';

describe('Album Page Test', () => {
  let wrapperContainer: HTMLElement;
  beforeEach(() => {
    const { container } = renderWithWrappers(<Album album={albumMock} />);
    wrapperContainer = container;
  });
  it('should render album in Album page', () => {
    expect(
      screen.getByText('One Hot Minute (Deluxe Edition)')
    ).toBeInTheDocument();
    expect(wrapperContainer.children[0]).toMatchSnapshot();
    expect(
      wrapperContainer.querySelector('.header__album--title').textContent
    ).toBe('One Hot Minute (Deluxe Edition)');
    expect(
      wrapperContainer.querySelector('.header__album--artist').textContent
    ).toBe('Red Hot Chili Peppers');
  });
  it('should have all song titles from the mock', () => {
    const allTracks = wrapperContainer.querySelectorAll(
      '.track--name.big-title'
    );
    allTracks.forEach((item, index) => {
      expect(item.textContent).toBe(albumMock.tracks.items[index].name);
    });
  });
});
