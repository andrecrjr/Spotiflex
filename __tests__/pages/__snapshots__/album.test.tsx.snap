// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album Page Test should render album in Album page 1`] = `
<section
  class="main--grid"
>
  <header
    class="header "
  >
    <div
      class="logo"
    >
      <span
        class="menu--icon"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M18.319 14.4326C20.7628 11.2941 20.542 6.75347 17.6569 3.86829C14.5327 0.744098 9.46734 0.744098 6.34315 3.86829C3.21895 6.99249 3.21895 12.0578 6.34315 15.182C9.22833 18.0672 13.769 18.2879 16.9075 15.8442C16.921 15.8595 16.9351 15.8745 16.9497 15.8891L21.1924 20.1317C21.5829 20.5223 22.2161 20.5223 22.6066 20.1317C22.9971 19.7412 22.9971 19.1081 22.6066 18.7175L18.364 14.4749C18.3493 14.4603 18.3343 14.4462 18.319 14.4326ZM16.2426 5.28251C18.5858 7.62565 18.5858 11.4246 16.2426 13.7678C13.8995 16.1109 10.1005 16.1109 7.75736 13.7678C5.41421 11.4246 5.41421 7.62565 7.75736 5.28251C10.1005 2.93936 13.8995 2.93936 16.2426 5.28251Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <form
        class="menu--option search--box"
      >
        <input
          placeholder="Search your favorite band, song, whatever"
          type="text"
        />
      </form>
      <a
        href="/"
      >
        Spotiflex
      </a>
    </div>
    <nav
      class="header__menu"
    >
      <ul
        class="header__menu--wrapper"
      >
        <li
          class="header__menu--item"
        >
          <a
            class="menu--item"
            href="/explorer"
          >
            <span
              class="menu--icon"
            >
              <svg
                color="rgb(148, 230, 26)"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(148, 230, 26);"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 10.9c-.61 0-1.1.49-1.1 1.1s.49 1.1 1.1 1.1c.61 0 1.1-.49 1.1-1.1s-.49-1.1-1.1-1.1zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm2.19 12.19L6 18l3.81-8.19L18 6l-3.81 8.19z"
                />
              </svg>
            </span>
            <span
              class="menu--option"
            >
              Explore
            </span>
          </a>
        </li>
        <li
          class="header__menu--item"
        >
          <a
            class="menu--item"
            href="/#"
            style="color: rgb(22, 22, 22);"
          >
            <svg
              color="#161616"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(22, 22, 22);"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                musics
              </title>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 3v10.55c-.59-.34-1.27-.55-2-.55-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4V7h4V3h-6z"
              />
            </svg>
            <span
              class="menu--option"
            >
              Songs
            </span>
          </a>
        </li>
        <li
          class="header__menu--item"
        >
          <a
            class="menu--item"
            href="/#"
            style="color: rgb(22, 22, 22);"
          >
            <span
              class="menu--icon"
            >
              <svg
                color="#161616"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(22, 22, 22);"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  podcast
                </title>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 14c1.66 0 2.99-1.34 2.99-3L15 5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 14 6.7 11H5c0 3.41 2.72 6.23 6 6.72V21h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
                />
              </svg>
            </span>
            <span
              class="menu--option"
            >
              Podcast
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <section
    class="main--wrapper"
  >
    <main
      class="main--block"
    >
      <div
        class="content fadeIn subpage"
      >
        <section
          class="header__album--wrapper"
        >
          <div
            class="header__album--pic"
          >
            <div
              style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
            >
              <img
                alt="One Hot Minute (Deluxe Edition)"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </div>
          </div>
          <div
            class="header__album--name"
          >
            <h2
              class="header__album--title"
            >
              One Hot Minute (Deluxe Edition)
            </h2>
            <div
              class="header__album--artist"
            >
              Red Hot Chili Peppers
              
            </div>
          </div>
        </section>
        <ul
          class="tracklist__wrapper"
        >
          <li
            class="tracklist__item"
            id="1eLAEzLXUu6SFHS5rSCrx1"
          >
            <span
              class="tracklist__item--play"
            >
              <section
                class="player--wrapper"
              >
                <span
                  class="player--control"
                  data-control="play"
                  role="button"
                >
                  <svg
                    class="button--play"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21ZM12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      d="M16 12L10 16.3301V7.66987L16 12Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </section>
            </span>
            <div
              class="tracklist__item--child"
            >
              <h4
                class="track--name big-title"
              >
                Warped
              </h4>
              <h5
                class="track--time"
              >
                05:04
              </h5>
            </div>
          </li>
          <li
            class="tracklist__item"
            id="0VLdJcQUsqHBBwqPp4CIKJ"
          >
            <span
              class="tracklist__item--play"
            >
              <section
                class="player--wrapper"
              >
                <span
                  class="player--control"
                  data-control="play"
                  role="button"
                >
                  <svg
                    class="button--play"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21ZM12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      d="M16 12L10 16.3301V7.66987L16 12Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </section>
            </span>
            <div
              class="tracklist__item--child"
            >
              <h4
                class="track--name big-title"
              >
                Aeroplane
              </h4>
              <h5
                class="track--time"
              >
                04:45
              </h5>
            </div>
          </li>
          <li
            class="tracklist__item"
            id="4VgylHaBo55VXqX4AiZiyc"
          >
            <span
              class="tracklist__item--play"
            >
              <section
                class="player--wrapper"
              >
                <span
                  class="player--control"
                  data-control="play"
                  role="button"
                >
                  <svg
                    class="button--play"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21ZM12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      d="M16 12L10 16.3301V7.66987L16 12Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </section>
            </span>
            <div
              class="tracklist__item--child"
            >
              <h4
                class="track--name big-title"
              >
                Deep Kick
              </h4>
              <h5
                class="track--time"
              >
                06:33
              </h5>
            </div>
          </li>
          <li
            class="tracklist__item"
            id="6HksxcBzMVdT17aq1Q79Xe"
          >
            <span
              class="tracklist__item--play"
            >
              <section
                class="player--wrapper"
              >
                <span
                  class="player--control"
                  data-control="play"
                  role="button"
                >
                  <svg
                    class="button--play"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21ZM12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      d="M16 12L10 16.3301V7.66987L16 12Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </section>
            </span>
            <div
              class="tracklist__item--child"
            >
              <h4
                class="track--name big-title"
              >
                My Friends
              </h4>
              <h5
                class="track--time"
              >
                04:08
              </h5>
            </div>
          </li>
          <li
            class="tracklist__item"
            id="2h3DA5sidcqjywLh36RFkU"
          >
            <span
              class="tracklist__item--play"
            >
              <section
                class="player--wrapper"
              >
                <span
                  class="player--control"
                  data-control="play"
                  role="button"
                >
                  <svg
                    class="button--play"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21ZM12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      d="M16 12L10 16.3301V7.66987L16 12Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </section>
            </span>
            <div
              class="tracklist__item--child"
            >
              <h4
                class="track--name big-title"
              >
                Coffee Shop
              </h4>
              <h5
                class="track--time"
              >
                03:08
              </h5>
            </div>
          </li>
          <li
            class="tracklist__item"
            id="6jR4FKWiil1FrsfEMt7jqu"
          >
            <span
              class="tracklist__item--play"
            >
              <section
                class="player--wrapper"
              >
                <span
                  class="player--control"
                  data-control="play"
                  role="button"
                >
                  <svg
                    class="button--play"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21ZM12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      d="M16 12L10 16.3301V7.66987L16 12Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </section>
            </span>
            <div
              class="tracklist__item--child"
            >
              <h4
                class="track--name big-title"
              >
                Pea
              </h4>
              <h5
                class="track--time"
              >
                01:47
              </h5>
            </div>
          </li>
          <li
            class="tracklist__item"
            id="0lK2JLm8XIl9MyY7gR9vSV"
          >
            <span
              class="tracklist__item--play"
            >
              <section
                class="player--wrapper"
              >
                <span
                  class="player--control"
                  data-control="play"
                  role="button"
                >
                  <svg
                    class="button--play"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21ZM12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      d="M16 12L10 16.3301V7.66987L16 12Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </section>
            </span>
            <div
              class="tracklist__item--child"
            >
              <h4
                class="track--name big-title"
              >
                One Big Mob
              </h4>
              <h5
                class="track--time"
              >
                06:02
              </h5>
            </div>
          </li>
          <li
            class="tracklist__item"
            id="5K9kCwY1e68H7JxU6893Uv"
          >
            <span
              class="tracklist__item--play"
            >
              <section
                class="player--wrapper"
              >
                <span
                  class="player--control"
                  data-control="play"
                  role="button"
                >
                  <svg
                    class="button--play"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21ZM12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      d="M16 12L10 16.3301V7.66987L16 12Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </section>
            </span>
            <div
              class="tracklist__item--child"
            >
              <h4
                class="track--name big-title"
              >
                Walkabout
              </h4>
              <h5
                class="track--time"
              >
                05:07
              </h5>
            </div>
          </li>
          <li
            class="tracklist__item"
            id="30vh2LXU2A9ccd0rgFYHpd"
          >
            <span
              class="tracklist__item--play"
            >
              <section
                class="player--wrapper"
              >
                <span
                  class="player--control"
                  data-control="play"
                  role="button"
                >
                  <svg
                    class="button--play"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21ZM12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      d="M16 12L10 16.3301V7.66987L16 12Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </section>
            </span>
            <div
              class="tracklist__item--child"
            >
              <h4
                class="track--name big-title"
              >
                Tearjerker
              </h4>
              <h5
                class="track--time"
              >
                04:19
              </h5>
            </div>
          </li>
          <li
            class="tracklist__item"
            id="6IshpbXajBj4iFny8FqmZF"
          >
            <span
              class="tracklist__item--play"
            >
              <section
                class="player--wrapper"
              >
                <span
                  class="player--control"
                  data-control="play"
                  role="button"
                >
                  <svg
                    class="button--play"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21ZM12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      d="M16 12L10 16.3301V7.66987L16 12Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </section>
            </span>
            <div
              class="tracklist__item--child"
            >
              <h4
                class="track--name big-title"
              >
                One Hot Minute
              </h4>
              <h5
                class="track--time"
              >
                06:23
              </h5>
            </div>
          </li>
          <li
            class="tracklist__item"
            id="4aAu1APORWioiUZOjUgmR4"
          >
            <span
              class="tracklist__item--play"
            >
              <section
                class="player--wrapper"
              >
                <span
                  class="player--control"
                  data-control="play"
                  role="button"
                >
                  <svg
                    class="button--play"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21ZM12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      d="M16 12L10 16.3301V7.66987L16 12Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </section>
            </span>
            <div
              class="tracklist__item--child"
            >
              <h4
                class="track--name big-title"
              >
                Falling into Grace
              </h4>
              <h5
                class="track--time"
              >
                03:48
              </h5>
            </div>
          </li>
          <li
            class="tracklist__item"
            id="29uoKpeQLuaCBAVp8oXCqR"
          >
            <span
              class="tracklist__item--play"
            >
              <section
                class="player--wrapper"
              >
                <span
                  class="player--control"
                  data-control="play"
                  role="button"
                >
                  <svg
                    class="button--play"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21ZM12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      d="M16 12L10 16.3301V7.66987L16 12Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </section>
            </span>
            <div
              class="tracklist__item--child"
            >
              <h4
                class="track--name big-title"
              >
                Shallow Be Thy Game
              </h4>
              <h5
                class="track--time"
              >
                04:33
              </h5>
            </div>
          </li>
          <li
            class="tracklist__item"
            id="25tFjJ6tVaDrGOw4SaT5BR"
          >
            <span
              class="tracklist__item--play"
            >
              <section
                class="player--wrapper"
              >
                <span
                  class="player--control"
                  data-control="play"
                  role="button"
                >
                  <svg
                    class="button--play"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21ZM12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      d="M16 12L10 16.3301V7.66987L16 12Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </section>
            </span>
            <div
              class="tracklist__item--child"
            >
              <h4
                class="track--name big-title"
              >
                Transcending
              </h4>
              <h5
                class="track--time"
              >
                05:46
              </h5>
            </div>
          </li>
          <li
            class="tracklist__item"
            id="7FGmrbu1GeHrxrr7lr15vH"
          >
            <span
              class="tracklist__item--play"
            >
              <section
                class="player--wrapper"
              >
                <span
                  class="player--control"
                  data-control="play"
                  role="button"
                >
                  <svg
                    class="button--play"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21ZM12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      d="M16 12L10 16.3301V7.66987L16 12Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </section>
            </span>
            <div
              class="tracklist__item--child"
            >
              <h4
                class="track--name big-title"
              >
                Let's Make Evil - 2006 Remaster
              </h4>
              <h5
                class="track--time"
              >
                05:17
              </h5>
            </div>
          </li>
          <li
            class="tracklist__item"
            id="3natnOToviL79uuVacIdUO"
          >
            <span
              class="tracklist__item--play"
            >
              <section
                class="player--wrapper"
              >
                <span
                  class="player--control"
                  data-control="play"
                  role="button"
                >
                  <svg
                    class="button--play"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21ZM12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      d="M16 12L10 16.3301V7.66987L16 12Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </section>
            </span>
            <div
              class="tracklist__item--child"
            >
              <h4
                class="track--name big-title"
              >
                Stretch You Out - 2006 Remaster
              </h4>
              <h5
                class="track--time"
              >
                06:18
              </h5>
            </div>
          </li>
          <li
            class="tracklist__item"
            id="6hMDMgqttllnWrN6g4vq9F"
          >
            <span
              class="tracklist__item--play"
            >
              <section
                class="player--wrapper"
              >
                <span
                  class="player--control"
                  data-control="play"
                  role="button"
                >
                  <svg
                    class="button--play"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21ZM12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      d="M16 12L10 16.3301V7.66987L16 12Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </section>
            </span>
            <div
              class="tracklist__item--child"
            >
              <h4
                class="track--name big-title"
              >
                Bob - 2006 Remaster
              </h4>
              <h5
                class="track--time"
              >
                05:43
              </h5>
            </div>
          </li>
        </ul>
      </div>
    </main>
  </section>
</section>
`;
